test/object-store/nested_collections.cpp

```diff
@@ -0,0 +1,175 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <catch2/catch_all.hpp>
+
+#include "util/test_file.hpp"
+#include <realm/object-store/object_schema.hpp>
+#include <realm/object-store/property.hpp>
+#include <realm/object-store/schema.hpp>
+
+#include <realm/db.hpp>
+#include <realm/collection_list.hpp>
+#include <realm/list.hpp>
+#include <realm/set.hpp>
+#include <realm/dictionary.hpp>
+
+using namespace realm;
+
+TEST_CASE("nested-list", "[nested-collections]") {
+    InMemoryTestFile config;
+    config.cache = false;
+    config.automatic_change_notifications = false;
+    auto r = Realm::get_shared_realm(config);
+    r->update_schema({
+        {"list_of_list", {{"nested_list", {PropertyType::Array, PropertyType::Array, PropertyType::Int}}}},
```

Comment:
I think it should rather be `{PropertyType::Array, PropertyType::Int | PropertyType::Array}`